src/mds/query.py

```diff
@@ -432,27 +337,34 @@ def get_clause(filter_dict):
 
         return add_filter(filter_dict["name"], filter_dict["op"], filter_dict["val"])
 
-    filter_tree = parse_filter(filter)
-    filter_dict = {}
-    if filter_tree:
-        filter_tree_visitor = FilterVisitor()
-        filter_dict = filter_tree_visitor.visit(filter_tree)
+    try:
+        filter_tree = None
+        if filter:
+            filter_tree = filter_param_grammar.parse(filter)
+        filter_dict = {}
+        if filter_tree:
+            filter_tree_visitor = FilterVisitor()
+            filter_dict = filter_tree_visitor.visit(filter_tree)
+    except (IncompleteParseError, ParseError, VisitationError):
+        raise HTTPException(
+            HTTP_400_BAD_REQUEST, f"filter URL query param syntax is invalid"
```

Comment:
maybe we can provide more details to the user? do the `IncompleteParseError, ParseError, VisitationError` return useful error messages?
if we don't want to return them to the user for some reason, i think we should at least log them to help us debug (maybe just `traceback.print_exc()`)